### PR TITLE
 Jump_Game in Java

### PR DESCRIPTION
**Pull request by Amanduttbhagat**
Fixes: #177

### CODE REVIEW

**Comment by Amanduttbhagat** [comment]
hello,
@anubhavitis @dheerajkotwani @tarun26091999 @todi-2000 @geekquad @plazzy99 @rudrakshi99
Can you please review my PR

**Comment by raksha009** [comment]
Plz mention issue no in description box. That would link the pr with issue automatically.

**Comment by Amanduttbhagat** [comment]
fixed #177

**Comment by Amanduttbhagat** [comment]
@raksha009 Is this right?

**Comment by raksha009** [comment]
Yeah, but you should mention the pr no in the description box. Pl take care from next time onwards :) Making the changes there for now.

Also, link your code with folder. Read guidelines.

**Comment by Amanduttbhagat** [comment]
Thank you @raksha009  I'll keep that in mind
I had linked code with folder
can you review it again

File: Code/Java/Jump_Game.java [hunk 1]
```diff
@@ -0,0 +1,121 @@
+/* Jump Game :
```
Comment (raksha009) [inline Code/Java/Jump_Game.java:1]: Pl. don't include * while writing multiline comments.

File: Code/Java/Jump_Game.java [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+ * We need to find the minimum number of jumps to reach the end of the array starting from the first element of the array,
+ * jumps value would be (-1) if we are not able to reach the end of the array
+ *
+ *
+ * example : array = {2,3,0,1,4,1,1,2,0,3,5}
```
Comment (raksha009) [inline Code/Java/Jump_Game.java:6]: Align properly

File: Code/Java/Jump_Game.java [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+				// Initialize checks again
+				checks = max_Jump_Reach - i;
+			}
+		}
+		return -1;
+	}
+
+	// main method
+	public static void main(String[] args) {
+		try {
```
Comment (raksha009) [inline Code/Java/Jump_Game.java:79]: Donot include try except block

File: Code/Java/Jump_Game.java [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+
+			System.out.println("minimum number of Jumps required : "+jumps);
+
+		}catch(Exception e) {
+
+		}
+	}
+}
+
+/* Test Cases :
```
Comment (raksha009) [inline Code/Java/Jump_Game.java:103]: Comment format!

File: Code/Java/Jump_Game.java [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+ *  2.  Input : 4
+ *              1 0 0 3
+ *     Output : minimum number of Jumps required : -1
+ *
+ *  3.  Input : 1
+ *              0
+ *     Output : minimum number of Jumps required : 0
+ *
+ *
+ * Time Complexity is O(n)
```
Comment (raksha009) [inline Code/Java/Jump_Game.java:117]: Specify n

**Review by raksha009 — CHANGES_REQUESTED** [review]

**Comment by Amanduttbhagat** [comment]
@raksha009 I have made the requested changes 
Can you please review it again

File: Code/Java/Jump_Game.java [hunk 6]
```diff
@@ -71,39 +71,39 @@ public static int jumpGame(int array[]) {
...
-   3.  Input : 1
-               0
-      Output : minimum number of Jumps required : 0
+ 3.  Input : 1
+             0
+    Output : minimum number of Jumps required : 0
 
 - Complexity Analysis :
-    * Time Complexity : O(n)
+ * Time Complexity : O(n)
```
Comment (raksha009) [inline Code/Java/Jump_Game.java:104]: Specify n not in the form of a para but after writing space and time complexity, you may just mention like
Where n is length of the input array or something similar :)

**Review by raksha009 — CHANGES_REQUESTED** [review]

**Comment by Amanduttbhagat** [comment]
@raksha009 can you review it again?

File: Code/Java/Jump_Game.java (hunk 6 above)
Comment (Amanduttbhagat) [reply Code/Java/Jump_Game.java:104]: @raksha009  I have a doubt that how can 'n' be specified as length or something else, as when we calculate time complexity we take number of iterations as a function for example f(n) = 5n + 13; and we take the highest degree term where 'n' is just a variable
Please correct me if I am wrong

Thank you

File: Code/Java/Jump_Game.java [hunk 7]
```diff
@@ -0,0 +1,110 @@
...
+		int[] array = new int[number];
+
+		// Putting values in array using for loop
+		for(int i = 0 ; i < array.length ; i++) {
+			array[i] = sc.nextInt();
+		}
+		// Initializing the value returned by jumpGame method to jumps
+		int jumps = JumpGame.jumpGame(array);
+
+		System.out.println("minimum number of Jumps required : "+jumps);
```
Comment (raksha009) [inline Code/Java/Jump_Game.java:86]: Add space around + operator. Rest LGTM!

**Review by raksha009 — CHANGES_REQUESTED** [review]

**Review by raksha009 — APPROVED** [review]

**Comment by Amanduttbhagat** [comment]
@shreyanspoddar can you please review this PR

Thank you

**Comment by Amanduttbhagat** [comment]
@shreyanspoddar @plazzy99 
are you there?

**Pull request merged**